homeassistant/components/apache_kafka/__init__.py

```diff
@@ -32,6 +36,9 @@
                 vol.Required(CONF_PORT): cv.port,
                 vol.Required(CONF_TOPIC): cv.string,
                 vol.Optional(CONF_FILTER, default={}): FILTER_SCHEMA,
+                vol.Optional(CONF_SECURITY_PROTOCOL, default="PLAINTEXT"): cv.string,
```

Comment:
What other values can be used here (other than `PLAINTEXT`)? This will allow anyone to input _any_ string.

Reply:
The problem is, is that Apache Kafka allows for the installation of additional/custom authentication providers.

I do think the name is incorrect/ambiguous. "Security Protocol" feels like telling something about the protocol used for data transfer, while this tells something about the authentication mechanism used. 

Maybe `CONF_AUTH_MECHANISM`, `CONF_AUTH_METHOD`, `CONF_AUTH_MODULE`, or something similar is more fitting.

Reply:
@bachya Agree. Thanks for catching that. I've tested `["PLAINTEXT", "SASL_SSL"]` options for non-auth, auth looks to work fine. LMK what you think of the changes.

@frenck I did have a thought over the name, but I settled for consistency over the [aiokafka docs](https://aiokafka.readthedocs.io/en/stable/api.html#aiokafka.AIOKafkaProducer), feel free to suggest changes - this is my first contribution and would like to adhere to the standards maintained by the community ;)